### PR TITLE
Merge/1.0 maint

### PR DESCRIPTION
**Pull request by enkore**
Many spurious conflicts, wonder what's up with that.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -9,7 +9,7 @@ deps =
      -rrequirements.d/development.txt
      -rrequirements.d/attic.txt
      -rrequirements.d/fuse.txt
-commands = py.test -rs --cov=borg --cov-config=.coveragerc --benchmark-skip --pyargs {posargs:borg.testsuite}
+commands = py.test -n 8 -rs --cov=borg --cov-config=.coveragerc --benchmark-skip --pyargs {posargs:borg.testsuite}
```
Comment (enkore) [inline tox.ini]: Could also use "-n auto" here.

**Comment by ThomasWaldmann** [comment]
Seems like we worked in parallel on that.
Yes, merging that was a PITA.
I'll review to see if we came out with the same stuff.

File: src/borg/archive.py [hunk 2]
```diff
@@ -1360,16 +1360,18 @@ def valid_item(obj):
             sort_by = sort_by.split(',')
             if any((first, last, prefix)):
                 archive_infos = self.manifest.archives.list(sort_by=sort_by, prefix=prefix, first=first, last=last)
+                if not archive_infos:
+                    logger.warning('--first/--last/--prefix did not match any archives')
```
Comment (ThomasWaldmann) [inline src/borg/archive.py]: i have a bit more here to handle less-than-expected cases for --first and --last.

File: tox.ini (hunk 1 above)
Comment (ThomasWaldmann) [inline tox.ini]: i used 2x corecount to make sure stuff is fully loaded.

File: src/borg/archive.py (hunk 2 above)
Comment (enkore) [inline src/borg/archive.py]: Picked this and the logger changes from your branch

**Comment by ThomasWaldmann** [comment]
tell me when you are finished taking the good pieces of mine.

**Pull request merged**